internal/engine/upper.go

```diff
@@ -1005,7 +1005,11 @@ func handleTiltfileLogAction(ctx context.Context, state *store.EngineState, acti
 	state.TiltfileCombinedLog = model.AppendLog(state.TiltfileCombinedLog, action, state.LogTimestamps, nil)
 }
 
-func handleAnalyticsOptAction(state *store.EngineState, action store.AnalyticsOptAction) {
+func handleAnalyticsOptAction(ctx context.Context, state *store.EngineState, action store.AnalyticsOptAction) {
+	// no logging on opt-out, because, well, opting out means the user just told us not to report data on them!
```

Comment:
nit (feel free to ignore): this comment feels mis-placed, because it's not about the clause it's on top of. consider:
```
if action.Opt...
    Incr...
}
// no logging on opt-out...
```

Reply:
I feel like it is even less about the line below the place you suggest, and your review comment would be even *more* applicable after that change!

I was thinking it's about the clause it's on top of since the clause could kind of similarly be `if action.Opt != analytics.OptOut` (except for OptDefault, which hopefully never appears here)

I guess I could move it above the `func` line entirely? but then a reader would expect it to describe the function as a whole.

how about changing "no logging on opt-out" to "only logging on opt-in"?

Reply:
fair nough, I trust your judgement!